content/articles/2017-07-ggplot2-3-0-0-tidy-evaluation.Rmarkdown

```diff
@@ -0,0 +1,116 @@
+
+```{r aes-quo}
+x_var <- quo(cyl)
+y_var <- quo(mpg)
+
+by_cyl <- mtcars %>%
+  group_by(!!x_var) %>%
+  summarise(mean = mean(!!y_var))
+
+ggplot(by_cyl, aes(!!x_var, mean)) +
+  geom_point()
+```
+
+You can now use [quasiquotation](https://adv-r.hadley.nz/quasiquotation.html) in [`aes()`](https://ggplot2.tidyverse.org/reference/aes.html), [`facet_wrap()`](https://ggplot2.tidyverse.org/reference/facet_wrap.html), and [`facet_grid()`](https://ggplot2.tidyverse.org/reference/facet_grid.html). For `aes()`, quasiquotation (`!!`, `!!!`, `:=`) replaces [`aes_()`](https://ggplot2.tidyverse.org/reference/aes_.html) and [`aes_string()`](https://ggplot2.tidyverse.org/reference/aes_.html) (though these functions are being soft deprecated, and will be around for a while).
+
+### Facetting with `vars()`
+
+To support quasiquotation in facetting we’ve added a new helper function: [`vars()`](https://ggplot2.tidyverse.org/reference/vars.html), short for variables. Instead of `facet_grid(x + y ~ a + b)` you can now write `facet_grid(vars(x, y), vars(a, b))`. The formula interface won’t go away; but the new `vars()` interface should be much easier to program with.
+
+`vars()` is used to supply variables or expressions, evaluated in the context of the dataset to form facetting groups.
+
+```{r facet-vars}
+p <- ggplot(mpg, aes(displ, cty)) + geom_point()
+
+p + facet_grid(rows = vars(drv))
+```
+
+Using quosures ensures that the variable comes from the context of the dataframe, as opposed to, say the global environment.
+
+```{r vars-env}
+mans <- c("chevrolet", "dodge", "ford", "toyota")
+
+year <- 2018
+
+mpg2 <- mpg %>%
+  filter(manufacturer %in% mans)
+
+d <- ggplot(mpg2, aes(displ, cty)) + geom_point()
+
+d + facet_grid(vars(year), vars(manufacturer))
+```
+
+Inside of `vars()` you can easily supply names, which will add titles to the facets.
+
+```{r labelled-grid}
+p + facet_grid(vars(Cylinder = cyl), labeller = label_both)
+```
+
+### Using `vars()` for wrapper functions
+
+`vars()` makes it easier to pass variables from wrapper functions.
+
+```{r wrap-by}
+p <- ggplot(mtcars, aes(wt, disp)) + geom_point()
+
+wrap_by <- function(...) {
+  facet_wrap(vars(...), labeller = label_both)
+}
+
+p + wrap_by(vs, am)
+```
+
+In our `wrap_by()` function above, we used tidy dots ([`...`](https://adv-r.hadley.nz/quasiquotation.html#dot-dot-dot-...)), which represent an arbitrary number of additional arguments. For a function to use *named arguments*, we'll need to "enquote" the named argument with [`enquo()`](http://rlang.r-lib.org/reference/quotation.html). To create a default name, we'll use [`quo_name()`](http://rlang.r-lib.org/reference/quo_label.html), which transforms a quosure into a simple string. Then we unquote and evaluate our arguments in their proper contexts using the [`!!`](http://rlang.r-lib.org/reference/quasiquotation.html) (read: bang bang) operator, and the `:=` operator to unquote the name.
```

Comment:
Can you explicit use "quote and unquote" phrase somewhere? That's how we're trying to get people to understand this.